src/gui/qgsrasterformatsaveoptionswidget.cpp

```diff
@@ -635,3 +635,28 @@ void QgsRasterFormatSaveOptionsWidget::showEvent( QShowEvent * event )
   QgsDebugMsg( "done" );
 }
 
+void QgsRasterFormatSaveOptionsWidget::setOptions( const QString& options )
+{
+  mOptionsTable->blockSignals( true );
+  mOptionsTable->clearContents();
+
+  QStringList values;
+  QStringList optionsList = options.trimmed().split( ' ', QString::SkipEmptyParts );
+  Q_FOREACH ( const QString &opt, optionsList )
+  {
+    int rowCount = mOptionsTable->rowCount();
+    mOptionsTable->insertRow( rowCount );
+
+    values = opt.split( '=' );
+    QTableWidgetItem* nameItem = new QTableWidgetItem( values.at( 0 ) );
+    mOptionsTable->setItem( rowCount, 0, nameItem );
+    QTableWidgetItem* valueItem = new QTableWidgetItem( values.at( 1 ) );
```

Comment:
Might be a good idea to check the length of values here - if a non KEY=VAL option is present this will crash. I'd check that values.count() == 2 and if not skip that option.